File: JasmineFastenerDepartment.Kotlin/app/src/main/java/com/szaytsev/jasminefastenerdepartment/network/models/SynchronizationResponse.kt
package com.szaytsev.jasminefastenerdepartment.network.models

import kotlinx.serialization.Serializable

@Serializable
data class SynchronizationResponse(
    val newProducts: List<SynchronizationProduct>,
    val modifiedProducts: List<SynchronizationProduct>,
    val historyEntries: List<SynchronizationHistoryEntry>
)

File: Jasmine.FastenerDepartment.Wpf/Jasmine.FastenerDepartment.Application/Models/Synchronization/SynchronizationResponse.cs
namespace Jasmine.FastenerDepartment.Application.Models.Synchronization;

/// <summary>
/// Synchronization response.
/// </summary>
public class SynchronizationResponse
{
    /// <summary>
    /// List of new synchronization products.
    /// </summary>
    public ICollection<SynchronizationProduct> NewProducts { get; set; }

    /// <summary>
    /// List of modified synchronization products.
    /// </summary>
    public ICollection<SynchronizationProduct> ModifiedProducts { get; set; }

    /// <summary>
    /// List of synchronization history entries.
    /// </summary>
    public ICollection<SynchronizationHistoryEntry> HistoryEntries { get; set; }
}
